things/api/things/http/requests.go

```diff
@@ -175,20 +186,20 @@ type createChannelsReq struct {
 
 func (req createChannelsReq) validate() error {
 	if req.token == "" {
-		return errors.ErrAuthentication
+		return apiutil.ErrBearerToken
 	}
 
 	if len(req.Channels) <= 0 {
-		return errors.ErrMalformedEntity
+		return apiutil.ErrEmptyList
 	}
 
 	for _, channel := range req.Channels {
-		if channel.ID != "" && validateUUID(channel.ID) != nil {
-			return errors.ErrMalformedEntity
+		if err := validateUUID(channel.ID); channel.ID != "" && err != nil {
```

Comment:
if `id == ""` wouldnt validate UUID fail anyways?

Reply:
Yes, but we want to check only if and ID was sent.

Reply:
I would also prefer having this separated, it's more readable:
```go
if channel.ID != "" {
    if err:=validateUUID(channel.ID); err != nil {
      return err
}
...
```